plugins/automation/lib/discourse_automation/scripts/remove_deleted_post_uploads.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# This script runs a revision to remove uploads that are attached to posts that have been deleted.
+# This automation is intended to be used with clean_up_uploads job.
+
+DiscourseAutomation::Scriptable.add(DiscourseAutomation::Scripts::REMOVE_DELETED_POST_UPLOADS) do
+  version 1
+
+  triggerables %i[recurring point_in_time]
+
+  script do |trigger, fields|
+    uploads_deleted_at = Time.now
+    edit_reason = I18n.t("discourse_automation.scriptables.remove_deleted_post_uploads.edit_reason")
+
+    Post
+      .with_deleted
+      .where.not(deleted_at: nil)
+      .joins(:upload_references)
+      .where("upload_references.target_type = 'Post'")
+      .each do |post|
+        if updated_raw =
+             post.raw.gsub!(%r{!\[([^\]|]+)(?:\|[^\]]*)?\]\(upload://([a-zA-Z0-9]+)[^)]+\)}, "")
```

Comment:
noting that this is the second time we use this regex:

https://github.com/discourse/discourse/blob/71834c898f2f3f5d11df3db6f9a5bab12acbccaa/plugins/discourse-ai/lib/completions/dialects/dialect.rb#L192-L199